robot_sort/robot_sort.py

```diff
@@ -81,32 +81,123 @@ def set_light_on(self):
+        # that way I can use conditions which lead to the None item being not-at-start become break triggers
+        # this will allow my program to simply error out?
+        # the alternative path is to use the None item as a position marker
+
+    def fizzselect(self):
+        if self.compare_item() > 0:
+            self.swap_item()
+
+    def move_to_end(self):
+        # reusable logic for movement loops
+        while True:
+            if self.can_move_right():
+                self.move_right()
+            else:
+                break
+
+    def move_to_start(self):
+        # reusable logic for movement loops
+        while True:
+            if self.can_move_left():
+                self.move_left()
+            else:
+                break
+
+    def seek_nothingness(self):
+        self.move_to_start()
+        # this loop allows us to seek the None item I'm using as a marker
+        while True:
+            if self.can_move_right() and self.compare_item() == None:
+                self.attain_nothingness()
+                break
+            elif self.can_move_right() == False and self.compare_item() == None:
+                self.set_light_on()
+                break
+            else:
+                if self.can_move_right():
+                    self.move_right()
+                else:
+                    break
+
+    def attain_nothingness(self):
+        self.swap_item()
+        self.move_right()
+        self.swap_item()
+        if self.can_move_right():
+            self.move_right()
+        else:
+            self.set_light_on()
+
+    def cleaning_the_end(self):
+        self.move_left()
+        if self.compare_item() < 0:
+            self.swap_item()
+            self.move_right()
+            self.swap_item()
+        else:
+            self.move_right()
+            self.swap_item()
+
     def sort(self):
         """
         Sort the robot's list.
         """
-        # Fill this out
+
+        # let's initialize this sorting routine with a quick swap and then nudge to the right?
+        # then proceed with a bubble sort? or the "fizzy" select sort`
+        self.swap_item()  # None at 0
+        self.move_right()
+
+        # sorting... start!
+        # Thinking about this a bit harder and writing a few methods
+        # I'm going to go with the None-placeholder
+        # That allows me to use None-at-end to mark "done"
+        # We can do a "comparison dance" at the end to make sure we're holding the None object
+
+        while True:
+            if self.can_move_left() == False:
+                if self.check_if_done() == True:
+                    return
+            else:
+                if self.light_is_on():
+                    self.cleaning_the_end()
+                    break
+                elif self.can_move_right():
+                    self.move_right()
+                    self.fizzselect()
+                else:
+                    self.seek_nothingness()
+
```

Comment:
Nice work getting the Robo-sort working.  Extracting functionality into the extra methods does help clean up the loop a bit, however there are quite a few methods and it gets hard to keep track of which are actually being used where, perhaps some are not being used and can be cleaned up, perhaps 1 or 2 might make sense to just keep inside other methods that only use them once.

All in all good job.  While you passed the tests for stretch you could still get the robots "time" for these down by about half by cleaning up some aspects of the algorithm.

Perhaps instead of defining the While loop as `while True:` you could consider making this a condition that is more meaningful.  (perhaps the status of the light could be used for this?)